components/View/index.jsx

```diff
@@ -76,13 +80,18 @@ class View extends Component {
    * @param {Object} nextProps The new component props.
    */
   componentWillReceiveProps(nextProps) {
-    if (nextProps.title !== this.props.title) {
-      this.props.setTitle(nextProps.title || this.props.title);
-    }
-
-    if (nextProps.viewTop && nextProps.viewTop !== this.props.viewTop) {
-      // Scroll to top
-      this.element.scrollTop = 0;
+    // Only run the logic, if the path for the View instance is active
+    if (nextProps.historyPathname === this.pathname) {
+      if ((nextProps.title !== this.props.title) ||
+        (this.props.navigatorTitle !== nextProps.title)) {
+        // Update the title if it distinguishes from the current navigator title
```

Comment:
`Update the title if it is different from the set navigator title.`

Reply:
done